### PR TITLE
Azure CMEK

### PR DESCRIPTION
**Pull request by fiquick**

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-384.surge.sh

Updated pages:
- [Encryption](https://neo4j-docs-aura-384.surge.sh/aura/platform/security/encryption)

**Comment by GuangxueCao** [comment]
Thanks for updating the doc 👍  

A few comments for consideration :) 

Screenshot 2024-07-02 at 11 18 02
Screenshot 2024-07-02 at 11 30 28

**Comment by ew-neo4j** [comment]
Adding my 2 cents:
1. Could we put the whole section of "Add Role Assignment" into one bullet point (with sub-bullet points) under "Create a key", as it's an optional step and more like a tickbox exercise rather than a key step to the flow?
2. For "Add role assignment to the Azure key" point 4, we should use a consistent term with the Console instructions for what the customer need to copy and paste:
- Neo4j CMK Application
- Neo4j CMK Application name
- Neo4j CMK Application ID
image

If we go for Neo4j CMK Application name, we should change the Console instructions as well.

@GuangxueCao - For role assignment, is there any other role that allows creating a key other than Key Vault Crypto Officer?
image

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-384.surge.sh

Updated pages:
- [Encryption](https://neo4j-docs-aura-384.surge.sh/aura/platform/security/encryption)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-384.surge.sh

Updated pages:
- [Encryption](https://neo4j-docs-aura-384.surge.sh/aura/platform/security/encryption)

**Review by GuangxueCao — APPROVED** [review]
LGTM!

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-384.surge.sh

Updated pages:
- [Encryption](https://neo4j-docs-aura-384.surge.sh/aura/platform/security/encryption)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-384.surge.sh

Updated pages:
- [Encryption](https://neo4j-docs-aura-384.surge.sh/aura/platform/security/encryption)

File: modules/ROOT/pages/platform/security/encryption.adoc [hunk 1]
```diff
@@ -99,4 +99,46 @@ In Aura, you can use AWS single-region keys, multi-region keys or replica keys a
...
+. When preparing to create a key, you might already have permissions, or you may need to grant a role assigment. This is one way to grant the relevant permissions to create a key:
+.. Inside the key vault, go to *Access Control (IAM)* and *add role assignment*.
+.. In the *Role* tab, select *Key Vault Administrator*.
+.. In the *Member* tab, select *User, group, or service principal*.
+.. *Select members* and select yourself or the relevant person, then *Review + Assign*.
+
+. Create a key in the Azure Key Vault. 
+. After the key is created, click into key version and copy the *Key Identifier*, you need it in the next step.
+. Go to *security settings* in the Aura Console and add a *Customer Managed Key*.
+. Follow the instructions in the Aura Console for the next sections:
```
Comment (AlexicaWright) [inline modules/ROOT/pages/platform/security/encryption.adoc:121]:
```suggestion
. Grant role assignment in the Azure key vault, if needed. 
. Create a key in the Azure key vault. 
. After the key is created, copy the *Key Identifier*, you need it in the next step.
. Go to *security settings* in the Aura Console and add a *Customer Managed Key*.
. Follow the instructions in the Aura Console for the next sections.
```

File: modules/ROOT/pages/platform/security/encryption.adoc [hunk 2]
```diff
@@ -99,4 +99,46 @@ In Aura, you can use AWS single-region keys, multi-region keys or replica keys a
...
+
+=== Create a service principal
+
+In the Azure Entra ID tenant where your key is located, create a service principal linked to the Neo4j CMK Application with the *Neo4j CMK Application ID* displayed in the Aura Console.
+
+One way to do this is by clicking the terminal icon at the top of the Azure portal, to open the Azure Cloud Shell.
+
+image::azurecloudshell.png[title = Azure Cloud Shell icon]
```
Comment (AlexicaWright) [inline modules/ROOT/pages/platform/security/encryption.adoc:129]:
```suggestion

```

File: modules/ROOT/pages/platform/security/encryption.adoc [hunk 3]
```diff
@@ -99,4 +99,46 @@ In Aura, you can use AWS single-region keys, multi-region keys or replica keys a
...
+. Create a key in the Azure Key Vault. 
+. After the key is created, click into key version and copy the *Key Identifier*, you need it in the next step.
+. Go to *security settings* in the Aura Console and add a *Customer Managed Key*.
+. Follow the instructions in the Aura Console for the next sections:
+
+=== Create a service principal
+
+In the Azure Entra ID tenant where your key is located, create a service principal linked to the Neo4j CMK Application with the *Neo4j CMK Application ID* displayed in the Aura Console.
+
+One way to do this is by clicking the terminal icon at the top of the Azure portal, to open the Azure Cloud Shell.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/platform/security/encryption.adoc:127]:
```suggestion

```

File: modules/ROOT/pages/platform/security/encryption.adoc [hunk 4]
```diff
@@ -99,4 +99,46 @@ In Aura, you can use AWS single-region keys, multi-region keys or replica keys a
...
+
+image::azurecloudshell.png[title = Azure Cloud Shell icon]
+
+Then, using https://learn.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest#az-ad-sp-create[Azure CLI] the command is:
```
Comment (AlexicaWright) [inline modules/ROOT/pages/platform/security/encryption.adoc:131]:
```suggestion
Using the Azure CLI, the command is: 
```

File: modules/ROOT/pages/platform/security/encryption.adoc [hunk 5]
```diff
@@ -99,4 +99,46 @@ In Aura, you can use AWS single-region keys, multi-region keys or replica keys a
...
+
+image::azurecloudshell.png[title = Azure Cloud Shell icon]
+
+Then, using https://learn.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest#az-ad-sp-create[Azure CLI] the command is: 
+
+[source,bash]
+----
+az ad sp create --id Neo4jCMKApplicationID
+----
+
```
Comment (AlexicaWright) [inline modules/ROOT/pages/platform/security/encryption.adoc:137]:
```suggestion
For more information about the Azure CLI, see link:https://learn.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest#az-ad-sp-create[az ad sp documentation].
```

File: modules/ROOT/pages/platform/security/encryption.adoc [hunk 6]
```diff
@@ -99,4 +99,46 @@ In Aura, you can use AWS single-region keys, multi-region keys or replica keys a
...
+=== Grant key permissions
+
+. To add role assignment to the Azure key, inside the key, go to *Access control (IAM)* and add *role assignment*.
+. In the *Role* tab, select *Key Vault Crypto Officer*.
+. In the *Member* tab, select *User, group, or service principal*.
+. *Select members* and paste the *Neo4j CMK Application name* that is displayed in the Aura Console. 
+. The *Neo4j CMK Application* should appear, select this application then *Review + Assign*.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/platform/security/encryption.adoc:144]:
```suggestion
To grant key permission to specific users, you can add role assignment from inside the Azure key.
```

**Review by AlexicaWright — COMMENTED** [review]
Some suggestion to minimize the level of detail about 3rd party UX.

File: modules/ROOT/images/azurecloudshell.png
[the remark this answers is not in the document]
Comment (fiquick) [reply modules/ROOT/images/azurecloudshell.png:1]: ok!

**Review by AlexicaWright — APPROVED** [review]

**Comment by fiquick** [comment]
thanks everyone! creating a fresh PR and merging :)

**Pull request closed without merging**